### DISCUSSION TITLE
Sorting by enum

### DISCUSSION THREAD

**Discussion by saraank** [ravendb/ravendb#13055] [Q&A]
What is the proper way to sort by the integer value of an enum?

**Comment by ayende** [comment]
There are a couple of options, first, you can manage it directly like so:
https://ayende.com/blog/195425-C/ravendb-custom-sorting-and-when-not-to-use-it?key=a1e334748f6846a1b198f97533d3b65a

You can also use: `store.Conventions.SaveEnumsAsIntegers = true;` to store enums as their numeric values.

The actual answer depend on what you need.

**Comment by saraank** [comment]
I have a problem changing the index because the collection has almost 16,000,000 documents and it will take a few hours to update this index.
The "SaveEnumAsIntegers" option is good but will affect all collections and interfere with data readings

**Reply by ayende** [reply]
I would recommend that you will change the index, to be honest, that is likely best.
Taking a few hours is fine, we do side by side building of the index, so won't impact ongoing queries.

**Reply by krivulcik** [reply]
Also, if the new code is incompatible with the old index (and vice versa), you can deploy the index in parallel to prevent issues with running unsupported queries, described here: https://github.com/ravendb/ravendb/discussions/12898
